odoo/addons/base/models/ir_property.py

```diff
@@ -113,21 +113,24 @@ def write(self, values):
         # if any of the records we're writing on has a res_id=False *or*
         # we're writing a res_id=False on any record
         default_set = False
-        if self._ids:
-            self.env.cr.execute(
-                'SELECT EXISTS (SELECT 1 FROM ir_property WHERE id in %s AND res_id IS NULL)', [self._ids])
-            default_set = self.env.cr.rowcount == 1 or any(
-                v.get('res_id') is False
-                for v in values
-            )
-        r = super(Property, self).write(self._update_values(values))
+
+        values = self._update_values(values)
+        default_set = (
+            # turning a record value into a fallback value
+            values.get('res_id') is False and any(record.res_id for record in self)
+        ) or any(
+            # changing a fallback value
+            not record.res_id and any(record[fname] != self._fields[fname].convert_to_record(value, self) for fname, value in values.items())
+            for record in self
+        )
+        r = super().write(values)
```

Comment:
leftover not related to this pr?

Reply:
This is in a specific commit, it avoid invalidating the ormcache when going on the settings page. (not a joke)